lib/cyclone-server/src/execution.rs

```diff
@@ -202,7 +175,14 @@ where
     }
 
     async fn child_send_function_request(stdin: ChildStdin, request: Request) -> Result<()> {
-        let value = serde_json::to_value(&request).map_err(ExecutionError::JSONSerialize)?;
+        let mut value = serde_json::to_value(&request).map_err(ExecutionError::JSONSerialize)?;
+
+        if let serde_json::Value::Object(ref mut map) = value {
+            map.insert(
+                "kind".to_string(),
+                serde_json::Value::String(request.kind().to_string()),
+            );
```

Comment:
I like this--I believe you managed to avoid changing the Veritech client's request types as a result. A nice spot to add it in, right before it's needed and only until it's needed.

Reply:
Yep, I want to limit the blast radius a bit and we can continue to use the request types as-is. Plus, having the the kind on the relevant trait means that we can't forget to implement it